sdk/spring-sdk/src/test/java/kalix/springsdk/testmodels/view/ViewTestModels.java

```diff
@@ -194,4 +216,36 @@ public User emptyState() {
       return null;
     }
   }
+
+  @Table(value = "users_view")
+  @Subscribe.ValueEntity(UserEntity.class)
+  public static class TransformMethodLackingEventParam extends View<TransformedUser> {
+
+    @Subscribe.ValueEntity(UserEntity.class)
+    public UpdateEffect<TransformedUser> onChange(TransformedUser user) {
+      return effects().updateState(new TransformedUser(user.name, user.email));
+    }
+
+    @Query("SELECT * FROM users_view WHERE email = :email")
+    @PostMapping("/users/by-email")
+    public TransformedUser getUserByEmail(@RequestBody ByEmail byEmail) {
+      return null;
+    }
+  }
+
+  @Table(value = "users_view")
+  @Subscribe.ValueEntity(UserEntity.class)
+  public static class TransformMethodWrongParamOrder extends View<TransformedUser> {
+
+    @Subscribe.ValueEntity(UserEntity.class)
+    public UpdateEffect<TransformedUser> onChange(User user, TransformedUser userView) {
+      return effects().updateState(new TransformedUser(userView.name, user.email));
+    }
+
+    @Query("SELECT * FROM users_view WHERE email = :email")
+    @PostMapping("/users/by-email")
+    public TransformedUser getUserByEmail(@RequestBody ByEmail byEmail) {
+      return null;
```

Comment:
I checked and this is working but I fail to understand how returning `null` can work here.

Reply:
The implementation on the Query methods is not used. The proxy responds to the resquests directly so the body of the methods are never actually used. We might have a way to allow the user to keep it abstract in the future but we don't at the moment.

Reply:
This method is never called, we only use it to tell the proxy how to create the index and how to respond. 

This is completely executed on the proxy side.

We will have to think how we want to solve this later. Users will be also puzzled by this.